itest/lnd_sendonion_test.go

```diff
@@ -94,3 +97,125 @@ func testSendOnion(ht *lntest.HarnessTest) {
 	// The invoice should show as settled for Dave.
 	ht.AssertInvoiceSettled(dave, invoices[0].PaymentAddr)
 }
+
+// testTrackOnion exercises the SwitchRPC server's TrackOnion endpoint,
```

Comment:
don't you think we can pack this itest into the sendonion one, they are so similar ?

Reply:
Yeah, originally was thinking I should do this also, but came up with a reason why it could be good to keep the separate. Since `TrackOnion` supports conditional error decryption, a dedicated test allows us to cleanly exercise both the server side and client side error decryption.

Check out this (and the linked) comment [here](https://github.com/lightningnetwork/lnd/pull/9489#discussion_r2270167952).

Reply:
Normally, I would expect a rpc client to just have the server decrypt forwarding errors so the client can handle them, but when the rpc client is the `ChannelRouter` type we need to use client side error decryption due to the way the function signature on `GetAttemptResult` is.

https://github.com/lightningnetwork/lnd/pull/9489#discussion_r2603546453